kube/src/api/admission.rs

```diff
@@ -0,0 +1,280 @@
+//! Contains types for implementing admission controllers.
+//!
+//! For more information on admission controllers, see:
+//! https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/
+//! https://kubernetes.io/blog/2019/03/21/a-guide-to-kubernetes-admission-controllers/
+
+use crate::{
+    api::{DynamicObject, GroupVersionKind, GroupVersionResource, Resource, TypeMeta},
+    Error,
+};
+
+use std::{collections::HashMap, convert::TryInto};
+
+use k8s_openapi::{
+    api::authentication::v1::UserInfo,
+    apimachinery::pkg::{apis::meta::v1::Status, runtime::RawExtension},
+};
+use serde::{Deserialize, Serialize};
+
+/// The top level struct used for Serializing and Deserializing AdmissionReview
+/// requests and responses.
+#[derive(Serialize, Deserialize, Clone, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct AdmissionReview<T: Resource> {
+    /// Contains the API version and type of the request.
+    #[serde(flatten, default)]
```

Comment:
i'm inclined to try without the default here initially.

Reply:
It's also possible that `AdmissionReview` should be an enum instead. We're not using the `TypeMeta` for anything, and we only ever use one of the types.

Reply:
Actually, they are needed in the response, nevermind!